ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/HeartbeatAgent.java

```diff
@@ -478,6 +447,40 @@ public HeartbeatAgent build(final KsqlEngine engine,
     }
   }
 
+  public static class HostStatus {
+    private boolean hostAlive;
+    private long lastStatusUpdateMs;
+
+    public HostStatus(
+        final boolean hostAlive,
+        final long lastStatusUpdateMs
+    ) {
+      this.hostAlive = hostAlive;
+      this.lastStatusUpdateMs = lastStatusUpdateMs;
+    }
+
+    public void setHostAlive(final boolean hostAlive) {
```

Comment:
Other threads (like during pull requests) are calling `isHostAlive` on this object after they call `getHostsStatus`, while your background thread is calling this method to update the value.

This technically isn't threadsafe, though it might work reasonably in practice.  You should either add synchronization (volatile, synchronized, etc) or make this object immutable.  I actually did the latter in my PR since I was reading the objects as well.  If you're fine with that, I can make that change and you can ignore this for now.

Reply:
I don't understand what the problem is. There is one thread that updates the value and the value is a boolean. There is no conditional updating (like checking the value and then update). What is the data corruption that can occur? The map is a `ConcurrentMap`,  new items are added only by one thread. Items are never removed. Only one thread updates values. There are multiple readers. But what corrupted value can the readers read? I am only updating the boolean (and the timestamp) of the object. 
I suppose it's fine to have immutable objects since there won't be many in the collection. I am just trying to understand whether it is necessary or if it is an overkill

Reply:
Tracked in #4394

Reply:
There's one writer and multiple readers in different threads of the boolean.  It's not that there will be some corrupted value in this scenario.  It's a binary bit that presumably is just false or true.  In java (and other languages), each thread has a cache of memory where local changes are made (just like cpu architectures), and they're only written back to main memory where other threads might read them when certain rules are followed (the java memory model).  In short, if you don't use volatile or synchronized, you're not guaranteed to ever read the written value across threads.

https://www.cs.umd.edu/~pugh/java/memoryModel/jsr-133-faq.html

I think you could fix this simply by making these fields volatile.  The immutability change would also work and just means that the `ConcurrentMap` is what does the job of synchronization.